File: src/com/dafttech/terra/game/world/tiles/TileFence.java
package com.dafttech.terra.game.world.tiles;

import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.dafttech.terra.game.world.entities.Entity;
import com.dafttech.terra.game.world.entities.Player;
import com.dafttech.terra.resources.Resources;

public class TileFence extends Tile {

    @Override
    public TextureRegion getImage() {
        return Resources.TILES.getImage("fence");
    }

    @Override
    public boolean canCollideWith(Entity entity) {
        return !(entity instanceof Player);
    }

    @Override
    public boolean isOpaque() {
        return false;
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/world/tiles/TileFence.scala
package com.dafttech.terra.game.world.tiles

import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.dafttech.terra.game.world.entities.Entity
import com.dafttech.terra.game.world.entities.living.Player
import com.dafttech.terra.resources.Resources
import monix.eval.Task

class TileFence extends Tile {
  override val getImage: Task[TextureRegion] = Resources.TILES.getImage("fence")

  override def isCollidableWith(entity: Entity): Boolean = !entity.isInstanceOf[Player]

  override def isOpaque: Boolean = false
}